### PR TITLE
Fixing multiple document updates

### LINKED ISSUE #7385: Documentation for KV DeleteKey operation seems incomplete
#### Overview of the Issue

[Documentation for the "Delete Key" operation in KV](https://www.consul.io/api/kv.html#delete-key) does not document the `dc` parameter.
Yet, when I try to do a request with the `dc` parameter, it seems to take it in account, so it seems to be a documentation typo.

#### Reproduction Steps

1. Run `curl -X DELETE localhost:8500/v1/kv/foo?dc=some-factice-datacenter`  
The output is: `No path to datacenter` which is expected, but not documented.

### LINKED ISSUE #7432: Broken Link to TLS Cipher Suites in Docs
The agent configuration docs on consul.io have a link to the source code on GitHub for showing the currently supported TLS cipher suites

https://www.consul.io/docs/agent/options.html#tls_cipher_suites

The source code link goes to here: https://github.com/hashicorp/consul/blob/master/tlsutil/config.go#L363

There are two issues. The first is that the link no longer points to our map of supported TLS cipher suites. Secondly we should probably make some way of preventing this from getting messed up again in the future.

### LINKED ISSUE #7646: API docs for config entries need updating to reflect newer types and ACL requirements
https://www.consul.io/api/config.html

Each section has a table of config entry types to ACL requirements. These tables are now out of date as they do not include service-resolver, service-router, service-splitter types and would also not include the upcoming terminating-gateway and ingress-gateway types.

We need to update these docs to make them complete.

### LINKED ISSUE #7663: Allow ACL tokens to be "Reloadable Configuration"
There seems to be a contradiction in the Vault best practices when using the Consul Secrets Engine and the fact that Consul doesn't allow reloading of tokens (https://www.consul.io/docs/agent/options.html#reloadable-configuration). Since you cannot just trigger a reload, you must restart the Consul process. Restarts of a consul server on an HA cluster may not be that bad but restarts of clients running on individual hosts where there is only one instance causes momentary outages.

#### Feature Description

We request that all the tokens in the ACL section of the config be part of Reloadable Configuration.

#### Use Case(s)

We have compliance requirements to rotate passwords and tokens periodically and Vault is perfect for that. The problem is when implementing a rotating Consul token, the tokens don't get reloaded with the HUP signal so Consul must be restarted which causes short downtime when using it for service discovery or as a KV datastore, etc.

### PR DESCRIPTION
**Pull request by jsosulska**
Fixes #7385 - Document KV Delete DC
Fixes #7432 - Updates Cipher Suites
Fixes #7646 - Config Entry types
Fixes #7663 - ACL Token Reloadable Docs
Updated Reft peers documentation to remove note about future information.

### CODE REVIEW

File: website/pages/docs/agent/options.mdx [hunk 1]
```diff
@@ -1804,7 +1804,9 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
 
 - `tls_cipher_suites` Added in Consul 0.8.2, this
   specifies the list of supported ciphersuites as a comma-separated-list. The list
-  of all supported ciphersuites is available in the [source code](https://github.com/hashicorp/consul/blob/master/tlsutil/config.go#L363).
+  of all supported ciphersuites is available in the source code
+  [here](https://github.com/hashicorp/consul/blob/master/tlsutil/config.go#L939)
```
[the remark this answers is not in the document]
Comment (jsosulska) [reply website/pages/docs/agent/options.mdx:1808]: Totally fine, and thanks for catching that :) I'll update!

**Comment by jsosulska** [comment]
Resolved your feedback Freddy. Gonna check the hyperlinks route correctly when the netlify preview is ready.

File: website/pages/docs/agent/options.mdx [hunk 2]
```diff
@@ -1930,13 +1931,15 @@ Reloading configuration does not reload all configuration items. The
 items which are reloaded include:
 
 - ACL Tokens
+- [Bootstrap](#config_entries_bootstrap)
```
Comment (freddygv) [inline website/pages/docs/agent/options.mdx:1934]:
```suggestion
- [Configuration Entry Bootstrap](#config_entries_bootstrap)
```

Comment (jsosulska) [reply website/pages/docs/agent/options.mdx:1934]: Will update, but can I ask what other types of bootstrapping there are that need this granularity?

Comment (dnephin) [reply website/pages/docs/agent/options.mdx]: There are at least 2 other bootstraps in Consul that I was able to find: one for electing an initial server leader, and another for ACLs

**Pull request merged**